### PR TITLE
Docs: Layout guidelines

### PR DESCRIPTION
**Pull request by hectoid**
### Summary
Added layout guidelines for mobile and tablet

#### What changed?

Added layout guidelines.
image
image

Moved Form layout code to the Forms section.
image

#### Why?

Designers ask for guidance on layout. We direct them to Lego guidelines in Dropbox. But Dropbox will soon be deprecated and many people already don't have access. So, we've moved them to Gestalt. The existing "layouts" page only had one layout for Forms, which makes more sense to include in the Forms section. We can revisit once we have sample code/templates for multiple layouts.

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-6422)
- [TDD](l[ink to Paper doc](https://paper.dropbox.com/doc/Lego-visual-language--CG348G4fLUFYO_J68DxZmm~NAg-16LVkb3NstMqJskFrXU22#:uid=803310587327265656450072&h2=Layout))
- [TDD update](https://docs.google.com/document/d/15oUsW2czdCtPC-G5X6RYpcx5R8pG7y36c6VaiGY_pPo/edit?usp=sharing)
- [Figma](https://www.figma.com/file/c7mskpSTbhp9NneVSdnJBj/Doc---Layout?type=design&node-id=1-9049&mode=design&t=EyGGMH05jRtREYYd-11)

### Checklist

- [x] Added unit and Flow Tests
- [x] Added documentation + accessibility tests
- [x] Verified accessibility: keyboard & screen reader interaction
- [x] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

**Comment by diyorbek** [comment]
LGTM

**Review by diyorbek — APPROVED** [review]

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
   "css.lint.validProperties": ["composes"],
   "editor.codeActionsOnSave": {
-    "source.fixAll.eslint": true,
-    "source.fixAll.shellcheck": true
+    "source.fixAll.eslint": "explicit",
```
Comment (rlingineni) [inline .vscode/settings.json:4]: @hectoid  - these vsocde changes seem to be back

**Comment by rlingineni** [comment]
@hectoid  - can you also add a [redirect](https://github.com/pinterest/gestalt/blob/319fed128eecc4ddb0fba68ad99f12c3bdd679ef/docs/redirects.js#L114) in this file since we're removing the other link

**Comment by hectoid** [comment]
> @hectoid - can you also add a [redirect](https://github.com/pinterest/gestalt/blob/319fed128eecc4ddb0fba68ad99f12c3bdd679ef/docs/redirects.js#L114) in this file since we're removing the other link

good catch @rlingineni ! added. also fixed the settings.js issues. I got a lot of errors locally this time around, so hopefully it's just an issue with vs code again. I'm not sure why that settings file keeps changing.

**Pull request merged**